songpal/device.py

```diff
@@ -269,16 +280,59 @@ async def activate_system_update(self) -> None:
 
     async def get_inputs(self) -> List[Input]:
         """Return list of available outputs."""
-        res = await self.services["avContent"]["getCurrentExternalTerminalsStatus"]()
-        return [
-            Input.make(services=self.services, **x)
-            for x in res
-            if "meta:zone:output" not in x["meta"]
-        ]
+        active_input_uri = ""
+        for z in await self.get_zones():
+            if z.active:
+                active_input_uri = (
+                    await self.get_available_playback_functions(output=z.uri)
+                )[0].uri
+                break
+
+        if (
+            "1.2"
+            in self.services["avContent"][
+                "getCurrentExternalTerminalsStatus"
+            ].supported_versions
+        ):
```

Comment:
Please add a `supports_version()` helper (or something similarly named) to the method class, this will help avoid nesting and to do something like this instead:
```
status_method = self.services["avContent"]["getCurrentExternalTerminalsStatus"]
if status_method.supports_version("1.2"):
    # call v1.2 method
else:
    # fallback
```

Reply:
Would be much cleaner.

```
method = self.services["avContent"]["getCurrentExternalTerminalsStatus"]
if method.supports_version("1.2"):
    method.use_version("1.2")
    res = await method({})
else:
    res = await method()
```